src/vs/workbench/parts/scm/electron-browser/scmViewlet.ts

```diff
@@ -549,7 +549,7 @@ class ResourceRenderer implements IListRenderer<ISCMResource, ResourceTemplate>
 
 		if (icon) {
 			template.decorationIcon.style.display = '';
-			template.decorationIcon.style.backgroundImage = `url('${icon}')`;
+			template.decorationIcon.style.backgroundImage = `url('${icon.toString(true)}')`;
```

Comment:
Please undo this change - it's coming from VSCode, and I verified this is the exact same in their codebase. It looks like they're using a function to read the URL value here

Reply:
the icon is in `URI` type, and it is class. 
This following code will only return `url('[object Object]')` string, and there is no way to parse it back to original `URI` object once it is converted to `'[object Object]'` string.
```
`url('${icon}')`
```

Reply:
Again, this code is in VSCode shipping right now. Are you going to tag with a SQL Carbon Edit and if so why? What use case does this fix / unblock?
- If this needs to be fixed and has visible user impact, we need to fix upstream. Next merge will lose your change anyhow.

Reply:
I'd suggest the fix here is a PR in VSCode codebase, happy to see you do that but this isn't the right place to fix...

Reply:
rolled back.